File: 2026/January/Day-01-Intersection-in-Y-Shaped-Lists/python/solution.py
class Solution:
    def intersectPoint(self, head1, head2):
        p1 = head1
        p2 = head2

        # Traverse until both pointers meet
        while p1 != p2:
            p1 = head2 if p1 is None else p1.next
            p2 = head1 if p2 is None else p2.next

        # Intersection node
        return p1

File: 2026/January/Day-01-Intersection-in-Y-Shaped-Lists/java/Solution.java
class Solution {
    public static Node intersectPoint(Node head1, Node head2) {
        Node p1 = head1;
        Node p2 = head2;

        // Traverse until both pointers meet
        while (p1 != p2) {
            p1 = (p1 == null) ? head2 : p1.next;
            p2 = (p2 == null) ? head1 : p2.next;
        }

        // Intersection node
        return p1;
    }
}
